kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/codec/ByteBufAccessorImpl.java

```diff
@@ -154,11 +145,9 @@ public int readUnsignedVarint() {
 
     @Override
     public ByteBuffer readByteBuffer(int length) {
-        // TODO use buf.nioBufferCount() and buf.nioBuffers() to avoid the copy if possible
-        ByteBuffer wrap = ByteBuffer.wrap(ByteBufUtil.getBytes(buf, buf.readerIndex(), length, false));
+        ByteBuffer byteBuffer = buf.internalNioBuffer(buf.readerIndex(), length);
```

Comment:
Couple of comments here:

1. `io.netty.buffer.ByteBuf#internalNioBuffer` is documented as internal use only.  I'm circumspect about disregarding that notice.

2. There's a change to semantics.  Previously the returned `ByteBuffer` belonged to the caller and the bytes contained were a copy.   The code is now exposing the underlying ByteBuffer object so if the caller calls say `ByteBuffer#mark`, might affect Netty's own operation.    If we did decide to do this, we should properly return a `ByteBuffer#slice` so the marks etc are independent.

3. I've got the same concerns about returning the live bytes rather than a copy.  Like I said above AFAIK this might be okay.